exercises/gigasecond/.meta/hints.md

```diff
@@ -0,0 +1,124 @@
+For example, for the above error you would add a file called `exercise_name.h`
+
+
+### Add the required declarations
+
+Running the tests a third time you see you will get new errors, similar to:
+
+```bash
+Compiling tests.out
+src/exercise_name.c:1:0: error: ISO C forbids an empty translation unit [-Werror=pedantic]
+cc1: all warnings being treated as errors
+test/test_exercise_name.c:13:1: error: unknown type name ‘bar_t’; did you mean ‘__bar_t’?
+    bar_t b = foo(a, 2.5);
+    ^~~~~~
+ __bar_t
+test/test_exercise_name.c:15:14: error: storage size of fizz isn’t known
+    struct baz fizz;
+               ^~~~
+test/test_exercise_name.c:29:20: error: implicit declaration of function ‘foo’ [-Werror=implicit-function-declaration]
+    bar_t b = foo_function(fizz, 2.5);
+```
+
+This error means that you need to add function, type and variable decarations to the header file to match those the test file is attempting to use.
+In doing this you will need to look at the build errors and the test code to determine what needs to be declared in the header file.
+
+For example, for the error above you should declare a type named  `bar_t`, a `struct` called `baz` and a function named `foo()`. 
+Additionally we can tell from looking at the above errors that function `foo()` has two parameters. 
+
+The first parameter is passed an argument of `fizz` and thus should have a type of `struct baz`.
+The second parameter is passed `2.5` and so could have one of two types, either `float` or `double`.
+You will need to look at the test code to determine which.
+
+The _names_ of the parameters are not determined by the test code and so are left up to you to decide, though in keeping with C programming practices they should be descriptive but not overly long.
+
+Further, we can see that the the return type expected by the test code is of the type `bar_t`.
+
+Putting this all together we end up with a function declaration that looks like the following:
+
+```c
+// The parameter names are not very good here they should be more descriptive in a real exercise.
+// We have decide on a double for the second parameter in this hypothetical example.
+bar_t foo(struct baz b, double d);
+```
+
+You should continue to do this for any further similar errors.
+To check that you have correctly determined the required declaration, just run the tests again and analyse any errors similarly.
+
+Additionally, remember to add any includes to the header file any headers that it itself requires.
+
+
+### Include guards
+
+Before you are finished with the header file, you should add include guards.
+Include guards are used to help prevent the file from being included multiple times by accident.
+If the file _was_ included multiple times then the functions and other items would be declared twice; In C this is an error because the compiler thinks you are trying to declare another _different_ function that has the same name.
```

Comment:
The `;` could probably just be a `.`, but I tried to take only a half step away from the em dash.